### PR TITLE
Add engineers constraint.

### PR DESCRIPTION
**Pull request by gantonayde**
Parse if an agent is an engineer when downloading the input. Add a min/max_engineers option to input.
Check if engineers on slot follow the distribution in the json input.

Change-type: patch

### CODE REVIEW

**Review by pranasziaukas — COMMENTED** [review]
Nice job Genadi.

My only concern is [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it).

As I understand, the main (and only) requirement for now is to have at least some engineering coverage on support at all times. This conceptually translates to
```python
any(agent["is_engineer"] for agent in agents) is True
```
or
```python
sum(agent["is_engineer"] for agent in agents) > 0
```
etc., which is less complex than introducing the notion of `min_engineers` and `max_engineers`.

If some additional complexity and performance concerns are not an actual issue then this may still be good.

Just my 2 cents.

File: lib/schemas/support-shift-scheduler-input.schema.json [hunk 1]
```diff
@@ -259,6 +265,16 @@
...
+              "min_engineers": {
+                "description": "Minimum number of engineers allowed for this time period",
+                "type": "number",
+                "minimum": 0
+              },
+              "max_engineers": {
+                "description": "Maximum number of engineers allowed for this time period",
+                "type": "number",
+                "minimum": 0
               }
```
[the remark this answers is not in the document]
Comment (gantonayde) [reply lib/schemas/support-shift-scheduler-input.schema.json:278]: In principle yes, but if we want to have 3 agents simultaneously, the `max_engineers` can be used to say "at least 1 agent needs to be non-engineer".

**Comment by gantonayde** [comment]
@pranasziaukas I have removed the max_engineers options as you suggested but the minimum needs to stay because the `any(...)` check implies at least 1, hence we hard-code the value and if the scheduler bot cannot find a feasible solution we are stuck. By having a `min_support_engineers` we can relax the constraint on-the-fly. 

The `any(...)` check will also cause problems for the devOps schedule, If someone there doesn't want to be alone on the balenaio channel. They simply won't be scheduled for devOps and it will be quite tricky to figure out why.

**Review by pranasziaukas — APPROVED** [review]
LGTM Genadi.

If we end up needing `max_engineers` soon then it means I was wrong.

Everything else makes sense and the code looks good 👍 Cheers.

**Pull request merged**